src/core/DNft.sol

```diff
@@ -161,30 +160,37 @@ contract DNft is ERC721Enumerable, Owned, IDNft {
     {
       _subDeposit(from, amount); 
       if (_collatRatio(from) < MIN_COLLATERIZATION_RATIO) { revert CrTooLow(); }
-      id2Withdrawn[from] += amount;
+      id2withdrawn[from] += amount;
       dyad.mint(to, amount);
       emit Withdrawn(from, to, amount);
   }
 
   /// @inheritdoc IDNft
   function redeemDyad(uint from, address to, uint amount)
     external 
-      isNftOwnerOrHasPermission(from)
+      isValidNft(from)
       rebase
     returns (uint) { 
-      id2Withdrawn[from] -= amount;
-      dyad.burn(msg.sender, amount); 
+      _burnDyad(from, amount);
       return _redeem(to, amount);
   }
 
+  function _burnDyad(uint from, uint amount)
+    private 
+    returns (uint) {
+      id2withdrawn[from] -= amount;
```

Comment:
Shouldn't this be +=? I might be off but if they are burning DYAD, then they are "unwithdrawing", right?

Reply:
Nevermind :)